### PR TITLE
Show mods and slots in new mod picker

### PR DESCRIPTION
**Pull request by ryan-rushton**
This won't be live in beta or the app.

image
image

### CODE REVIEW

File: src/app/loadout-builder/generated-sets/GeneratedSetSockets.tsx [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+import { PlugCategoryHashes } from 'data/d2/generated-enums';
+import { DestinyInventoryItemDefinition } from 'bungie-api-ts/destiny2';
+import SocketDetails from '../../item-popup/SocketDetails';
+import ReactDOM from 'react-dom';
+import './GeneratedSetSockets.scss';
+
+const undesireablePlugs = [
+  PlugCategoryHashes.ArmorSkinsEmpty,
+  PlugCategoryHashes.Shader,
+  PlugCategoryHashes.ArmorSkinsEmpty,
```
Comment (bhollis) [inline src/app/loadout-builder/generated-sets/GeneratedSetSockets.tsx:15]: Got some dups in here?

File: src/app/loadout-builder/generated-sets/GeneratedSetSockets.scss [hunk 2]
```diff
@@ -0,0 +1,44 @@
+@import '../../variables.scss';
+
+.lockedItems {
```
Comment (bhollis) [inline src/app/loadout-builder/generated-sets/GeneratedSetSockets.scss:3]: I'm generally a fan of new stuff being CSS modules, just a note for later

File: src/app/loadout-builder/generated-sets/GeneratedSetMod.tsx [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+      style={{ gridColumn }}
+      title={plugDef.displayProperties.name}
+      tabIndex={0}
+      onClick={onClick}
+    >
+      <BungieImage className="item-img" src={plugDef.displayProperties.icon} />
+      {energyCostElementOverlay && (
+        <>
+          <div
+            style={{ backgroundImage: `url("${bungieNetPath(energyCostElementOverlay)}")` }}
```
Comment (bhollis) [inline src/app/loadout-builder/generated-sets/GeneratedSetMod.tsx:36]: There's a `bungieBackgroundStyle` helper

File: src/app/loadout-builder/generated-sets/GeneratedSetSockets.scss (hunk 2 above)
Comment (ryan-rushton) [reply src/app/loadout-builder/generated-sets/GeneratedSetSockets.scss:3]: Yeah same, this says its a new file even though its not it just got renamed and small modifications. There was a reason why it wasn't a module but I can't remember why. I might try convert it.

File: src/app/loadout-builder/generated-sets/GeneratedSetMod.tsx (hunk 3 above)
Comment (ryan-rushton) [reply src/app/loadout-builder/generated-sets/GeneratedSetMod.tsx:36]: K will check it

File: src/app/loadout-builder/generated-sets/GeneratedSetSockets.tsx (hunk 1 above)
Comment (ryan-rushton) [reply src/app/loadout-builder/generated-sets/GeneratedSetSockets.tsx]: Fixed

File: src/app/loadout-builder/generated-sets/GeneratedSetSockets.scss (hunk 2 above)
Comment (ryan-rushton) [reply src/app/loadout-builder/generated-sets/GeneratedSetSockets.scss]: Converted it over

File: src/app/loadout-builder/generated-sets/GeneratedSetMod.tsx (hunk 3 above)
Comment (ryan-rushton) [reply src/app/loadout-builder/generated-sets/GeneratedSetMod.tsx]: Done

**Pull request merged**